orchard_poseidon.py

```diff
@@ -0,0 +1,299 @@
+from orchard_pallas import Fp
+import numpy as np
+
+# Number of full rounds
+R_F = 8
+# Number of partial rounds
+R_P = 58
+# Width
+t = 3
+
+# Initial capacity element
+CAPACITY_ELEMENT = Fp(1 << 65)
+
+def poseidon_hash(x, y):
+    return perm([x, y, CAPACITY_ELEMENT])
```

Comment:
Move these into `orchard_poseidon_hash.py` and then add test vector generation (e.g. some sequential inputs and some random) both here for `perm` and in `orchard_poseidon_hash.py` for `poseidon_hash`.